system-addon/test/functional/mochitest/browser_asrouter_targeting.js

```diff
@@ -0,0 +1,68 @@
+ChromeUtils.defineModuleGetter(this, "ASRouterTargeting",
+"resource://activity-stream/lib/ASRouterTargeting.jsm");
+ChromeUtils.defineModuleGetter(this, "ProfileAge",
+  "resource://gre/modules/ProfileAge.jsm");
+
+// ASRouterTargeting.isMatch
+add_task(async function should_do_correct_targeting() {
+  is(await ASRouterTargeting.isMatch("FOO", {FOO: true}), true, "should return true for a matching value");
+  is(await ASRouterTargeting.isMatch("!FOO", {FOO: true}), false, "should return false for a non-matching value");
+});
+
+add_task(async function should_handle_async_getters() {
+  const context = {get FOO() { return Promise.resolve(true); }};
+  is(await ASRouterTargeting.isMatch("FOO", context), true, "should return true for a matching async value");
+});
+
+// ASRouterTargeting.findMatchingMessage
+add_task(async function find_matching_message() {
+  const messages = [
+    {id: "foo", targeting: "FOO"},
+    {id: "bar", targeting: "!FOO"}
+  ];
+  const context = {FOO: true};
+
+  const match = await ASRouterTargeting.findMatchingMessage(messages, context);
+
+  is(match, messages[0], "should match and return the correct message");
+});
+
+add_task(async function return_nothing_for_no_matching_message() {
+  const messages = [{id: "bar", targeting: "!FOO"}];
+  const context = {FOO: true};
+
+  const match = await ASRouterTargeting.findMatchingMessage(messages, context);
+
+  ok(typeof match === "undefined", "should match and return the correct message");
```

Comment:
nit: I think you can do `is(match, undefined, "should match and return the correct message")` here. Just for consistency and clarity